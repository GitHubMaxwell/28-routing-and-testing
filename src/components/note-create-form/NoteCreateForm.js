import React from 'react'

// onComplete the NoteForm should add a note to the application state

class NoteCreateForm extends React.Component {
    constructor(props) {
        super(props)
        this.formComplete = this.formComplete.bind(this);
        this.handleTitle = this.handleTitle.bind(this)
        this.handleContent = this.handleContent.bind(this)
    }

    formComplete(event) {
        event.preventDefault();
        // console.log(event.target.name)
        this.props.addNote()
        document.getElementById('form').reset()
    }

    handleTitle(e) {
        this.props.updateTitle(e.target.value)
    }
    handleContent(e) {
        this.props.updateContent(e.target.value)
    }
    render() {
        return (
            <React.Fragment>
                <h2>Create Note</h2>
                <form id="form" onSubmit={this.formComplete}>
                
                <label>Note Title:</label>
                <input type="text" onChange={this.handleTitle}/>
                <label>Note Content:</label>
                <textarea onChange={this.handleContent}></textarea>
                <button>Submit</button>

                </form>

            </React.Fragment>
        )
    }
}

export default NoteCreateForm;