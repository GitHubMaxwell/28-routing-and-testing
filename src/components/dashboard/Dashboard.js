import React from 'react'
import NoteList from '../note-list/NoteList.js'
import NoteCreateForm from '../note-create-form/NoteCreateForm.js'
import uuidv1 from 'uuid/v1'
/*

    The Dashboard component should manage the entire application state
    The state should contain a notes array
    It should have a bound addNote(note) method that adds a note to state.notes
        each note that gets added should have the following data
            id: always should contain the result of uuid.v1()
            editing: false by default
            completed: false by default
            content: user provided content
            title: user provided title
    It should have a bound removeNote(note) method that removes a note from state.notes based on its id

*/

class Dashboard extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            note: {
                id: '',
                editing: false,
                completed: false,
                content: '',
                title: '',
            },
            notes: [{
                id: 'fakeID',
                editing: false,
                completed: false,
                content: 'this is the content',
                title: 'TITLE',
            }]
        }
        this.addNote = this.addNote.bind(this)
        this.removeNote = this.removeNote.bind(this)
        this.updateNote = this.updateNote.bind(this)
    }

    updateNote(event) {
        let editing, completed = false;
        let id = uuidv1();
        console.log(this.state.note)

        this.setState = {
            note: {
                id,
                title: event.target.value,
                content: event.target.value,
                editing,
                completed
            }
        }
    }

    addNote() {
        // event.preventDefault();

        /*
        the object equivalent

        addItem(data) {
        let item = {};
        item[data.id] = data.text;
        this.setState( Object.assign(this.state.items,item) );
        }

        */
        // not a function?
        this.setState((prevState) => {
            return { note: {
                // id : uuidv1(),
                editing : !prevState.editing,
                completed : !prevState.completed
            }
          };
        });

        console.log('SUBMIT', this.state.note)

        // let notes = this.state.notes.push(this.state.note)
        // console.log('ARRAY Before', notes)
        console.log('ARRAY state', this.state.notes)


        this.setState((prevState) => {
            return { notes : prevState.notes.push(this.state.note)};
        });

        // console.log('ARRAY After', this.state.notesArr)
    }

    removeNote(note) {
        // removes a note from state.notes based on its id
        // might not have to do note.id but just note
        console.log('removeNote')
        // let index = this.state.notes.indexOf(note.id)
        // console.log(index)
        // let notes = this.state.notes.splice(index, 1)
        let notes = this.state.notes.splice(0, 1)

        console.log(notes)

        this.setState({
            notes
        })
        console.log('DELETE after',notes)

    }

    /*
    do this 

        componentDidUpdate(prevProps) {
            // Typical usage (don't forget to compare props):
            if (this.props.userID !== prevProps.userID) {
                this.fetchData(this.props.userID);
            }
        }

    */

    //might need to send attribute with addNote?
    render() {
        return (
            <React.Fragment>
                <h2>Dashboard</h2>
                <NoteCreateForm updateNote={this.updateNote} addNote={this.addNote} />
                <NoteList notesArr={this.state.notes} removeNote={this.removeNote} />
            </React.Fragment>
        )
    }
}

export default Dashboard;