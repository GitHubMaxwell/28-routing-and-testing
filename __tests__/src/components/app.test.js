import React from 'react'
import renderer from 'react-test-renderer'
import App from '../../../src/components/app/App.js'
import Dashboard from '../../../src/components/dashboard/Dashboard.js'
import NoteCreateForm from '../../../src/components/note-create-form/NoteCreateForm.js'


xdescribe('Proof of Life Test', () => {
    it('POL', () => {
        let app = shallow(<App/>);
        expect(app.find('main').exists()).toBeTruthy();
    })
})

describe('<Dashboard/> and <NoteCreateForm/> Initial State Snapshot Test', () => {
    it('Dashboard Snapshot Test', () => {
        const component = renderer.create(
            <Dashboard/>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
    it('NoteCreateForm Snapshot Test', () => {
        const component = renderer.create(
            <NoteCreateForm/>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})