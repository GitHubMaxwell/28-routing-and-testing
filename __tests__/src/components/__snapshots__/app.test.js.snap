// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard/> and <NoteCreateForm/> Initial State Snapshot Test Dashboard Snapshot Test 1`] = `
Array [
  <h2>
    Dashboard
  </h2>,
  <h2>
    Create Note
  </h2>,
  <form
    id="form"
    onSubmit={[Function]}
  >
    <label>
      Note Title:
    </label>
    <input
      onChange={[Function]}
      type="text"
    />
    <label>
      Note Content:
    </label>
    <textarea
      onChange={[Function]}
    />
    <button>
      Submit
    </button>
  </form>,
  <h2>
    Note List
  </h2>,
  <ul />,
]
`;

exports[`<Dashboard/> and <NoteCreateForm/> Initial State Snapshot Test NoteCreateForm Snapshot Test 1`] = `
Array [
  <h2>
    Create Note
  </h2>,
  <form
    id="form"
    onSubmit={[Function]}
  >
    <label>
      Note Title:
    </label>
    <input
      onChange={[Function]}
      type="text"
    />
    <label>
      Note Content:
    </label>
    <textarea
      onChange={[Function]}
    />
    <button>
      Submit
    </button>
  </form>,
]
`;
